plugins/docsettingtweak.koplugin/main.lua

```diff
@@ -98,7 +98,13 @@ function DocSettingTweak:onDocSettingsLoad(doc_settings, document)
                 doc_settings.data = util.tableDeepCopy(directory_defaults:readSetting(directory))
                 break
             else
-                directory = FFIUtil.dirname(directory)
+                next_directory = FFIUtil.dirname(directory)
+                if directory == next_directory then
```

Comment:
good catch i forgot to test the `/` or `.` case.
there is no need for this:
```
next_directory = FFIUtil.dirname(directory)
```

```suggestion
                if directory == "/" or  directory == "." then
                    -- have reached the root, abort
```